front_end/panels/ai_chat/mcp/MCPMetaTools.ts

```
@@ -0,0 +1,204 @@
+
+    // Choose ranking strategy
+    const strategy = args.strategy || 'heuristic';
+    let rankedTools: Array<{ key: string; name: string; description: string; serverId: string }>;
+
+    switch (strategy) {
+      case 'llm':
+        rankedTools = await this.rankWithLLM(candidates, args.query, k);
+        break;
+      case 'hybrid':
+        // For hybrid: use heuristic pre-filtering then LLM ranking on top candidates
+        const prefilterLimit = Math.min(candidates.length, k * 5); // 5x expansion for LLM to choose from
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)